diff --git a/train.py b/train.py
index 090e248..ac67156 100644
--- a/train.py
+++ b/train.py
@@ -25,7 +25,7 @@ def train(
     pooler_idx: int = 0, 
     load_state_dict: str = None,
     data_root: str = Config.Train,
-    tokenization_type: str = PreProcessType.Base,
+    preprocess_type: str = PreProcessType.Base,
     epochs: int = Config.Epochs,
     valid_size: float = Config.ValidSize,
     train_batch_size: int = Config.Batch32,
@@ -43,7 +43,7 @@ def train(
 
     # load data
     dataset = REDataset(
-        root=data_root, tokenization_type=tokenization_type, device=device
+        root=data_root, preprocess_type=preprocess_type, device=device
     )
     if valid_size == 0:
         is_valid = False # validation flag
@@ -246,7 +246,7 @@ if __name__ == "__main__":
     parser.add_argument("--pooler-idx", type=int, default=0)
     parser.add_argument("--load-state-dict", type=str, default=LOAD_STATE_DICT)
     parser.add_argument("--data-root", type=str, default=Config.Train)
-    parser.add_argument("--tokenization-type", type=str, default=PreProcessType.Base)
+    parser.add_argument("--preprocess-type", type=str, default=PreProcessType.ES)
     parser.add_argument("--epochs", type=int, default=Config.Epochs)
     parser.add_argument("--valid-size", type=int, default=Config.ValidSize)
     parser.add_argument("--train-batch-size", type=int, default=Config.Batch8)
diff --git a/wandb/latest-run b/wandb/latest-run
index ca43bec..db1fe88 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20210417_091840-7dmyj8nv
\ No newline at end of file
+run-20210418_070913-2hb9ksrd
\ No newline at end of file
