diff --git a/train.py b/train.py
index dec525c..6787d61 100644
--- a/train.py
+++ b/train.py
@@ -284,7 +284,7 @@ if __name__ == "__main__":
     parser.add_argument("--valid-batch-size", type=int, default=512)
     parser.add_argument("--optim-type", type=str, default=Optimizer.Adam)
     parser.add_argument("--loss-type", type=str, default=Loss.CE)
-    parser.add_argument("--lr", type=float, default=Config.LRSlower)
+    parser.add_argument("--lr", type=float, default=Config.LRSlow)
     parser.add_argument("--lr-scheduler", type=str, default=Optimizer.CosineAnnealing)
     parser.add_argument("--device", type=str, default=Config.Device)
     parser.add_argument("--seed", type=int, default=Config.Seed)
diff --git a/wandb/latest-run b/wandb/latest-run
index f1bf844..397bf72 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20210418_123951-2v936pyt
\ No newline at end of file
+run-20210418_150205-3v4qkwgd
\ No newline at end of file
