diff --git a/config.py b/config.py
index c60715f..703a33c 100644
--- a/config.py
+++ b/config.py
@@ -26,7 +26,7 @@ class Config:
     SavePath: str = SAVEPATH if os.path.isfile(SAVEPATH) else DOT + SAVEPATH
     Logs: str = LOGS if os.path.isfile(LOGS) else DOT + LOGS
     NumClasses: int = 42
-    Epochs: int = 1
+    Epochs: int = 20
     Batch32: int = 32
     Batch64: int = 64
     LR: float = 0.001
diff --git a/wandb/latest-run b/wandb/latest-run
index 5b0a4e2..25c39ef 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20210416_190318-m4rz970f
\ No newline at end of file
+run-20210416_190945-14py9lif
\ No newline at end of file
